### PR TITLE
Service Accounts: Allow unsetting token expiry date

### LINKED ISSUE #65085: Chore: Display `No Expiration` toggle for SAT expiry date
**What is the chore?**:
Issue spawned of #64893

The Service Account Token expiry date toggle disappears when trying to set a date and the option to go back to the `No expiration` date disappears.

The overall result should look as outlined by the following [comment](https://github.com/grafana/grafana/pull/64893#issuecomment-1476217575).

### PR DESCRIPTION
**Pull request by kousikmitra**
**What is this feature?**

This fix will allow the users to go back to `no expiration` after clicking on `set expiration date`

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #65085 

grafana-sat

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/serviceaccounts/components/CreateTokenModal.tsx [hunk 1]
```diff
@@ -109,7 +109,7 @@ export const CreateTokenModal = ({ isOpen, token, serviceAccountLogin, onCreateT
               }}
             />
           </Field>
-          {!isWithExpirationDate && (
+          {
```
Comment (IevaVasiljeva) [inline public/app/features/serviceaccounts/components/CreateTokenModal.tsx:112]: nit: I think you can remove the curly braces here and on line 121.

**Review by IevaVasiljeva — APPROVED** [review]
Thanks for your contribution! I've tested the PR and all works as expected 🚢

**Comment by IevaVasiljeva** [comment]
@kousikmitra I've added the required labels (it's required to set PR milestone and information about backports and change-log before merging: https://github.com/grafana/grafana/blob/main/contribute/merge-pull-request.md#before-merge), please double-check if they look good to you :)

If all looks good, please sign the CLA (https://github.com/grafana/grafana/pull/65862#issuecomment-1494580116) and we can then get your PR merged.

File: public/app/features/serviceaccounts/components/CreateTokenModal.tsx (hunk 1 above)
Comment (kousikmitra) [reply public/app/features/serviceaccounts/components/CreateTokenModal.tsx]: Fixed

**Comment by kousikmitra** [comment]
@IevaVasiljeva Signed the CLA, I guess it's good to go.

**Pull request merged**